src/Mod/Path/PathScripts/PathUtils.py

```diff
@@ -718,7 +722,30 @@ def __get_depths(self, equalstep=False):
             )[1:]
 
         depths.reverse()
-        return depths
+
+        if len(depths) < 2:
+            return depths
+
+        return self.__filter_roughly_equal_depths(depths)
+
+    def __filter_roughly_equal_depths(self, depths):
+        """Depths arrive sorted from largest to smallest, positive to negative.
+        Return unique list of depths, using PathGeom.isRoughly() method to determine
+        if the two values are equal.  Only one of two consecutive equals are removed.
+
+        The assumption is that there are not enough consecutively roughly-equal depths
+        to be removed, so as to eliminate an effective step-down depth with the removal
+        of repetitive roughly-equal values."""
+
+        # Do not need a copy of original `depths` because we are only reading and copying
+        depths.sort()  # Sort again in case user_depths were not sorted
```

Comment:
A good practice is to NEVER modify directly the arguments passed into the function. 
sort() changes the depths list.  It actually modifies the current list.  Doing so can cause unexpected consequences, especially if the code is later modified to work in a multi-threaded way.  This is one of the bedrock principles of functional programming and while we aren't trying to write functional code, it's an easy thing to implement that makes the code more reliable.

This case is low risk, but you could imagine that another thread is simultaneously processing the list for another purpose.  Sorting the list here could break the other thread and it would be very difficult to debug because the bug would not be consistent.